ColorfulCLI/main.swift

```diff
@@ -81,27 +81,20 @@ func threeColorPermutations (for graph : ColoredGraph) -> [ColoredGraph] {
     
     // for speed
     let permutations = [
-        [Colors.red, Colors.grn, Colors.blu],
-        [Colors.red, Colors.blu, Colors.grn],
-        [Colors.grn, Colors.red, Colors.blu],
-        [Colors.grn, Colors.blu, Colors.red],
-        [Colors.blu, Colors.red, Colors.grn],
-        [Colors.blu, Colors.grn, Colors.red]
+        [Colors.first, Colors.second, Colors.third],
+        [Colors.first, Colors.third, Colors.second],
+        [Colors.second, Colors.first, Colors.third],
+        [Colors.second, Colors.third, Colors.first],
+        [Colors.third, Colors.first, Colors.second],
+        [Colors.third, Colors.second, Colors.first]
     ]
     
     var graphPermutations = [ColoredGraph]()
     
     for permutation in permutations {
         var newGraph = graph
         for (vertexID, (color, _)) in newGraph {
-            switch color {
-            case .red:
-                newGraph[vertexID]!.0 = permutation[0]
-            case .grn:
-                newGraph[vertexID]!.0 = permutation[1]
-            case .blu:
-                newGraph[vertexID]!.0 = permutation[2]
-            }
+            newGraph[vertexID]!.0 = permutation[color]
```

Comment:
This won't work. Would have to be `color.rawValue`.
Furthermore, I _really_ don't like depending on the raw value of an enum being a valid index into an array...
We could change the hardcoded permutations to be a map of \[Colors: Colors] instead, and it'd allow you to keep this fairly idiomatic.
Remember when rewriting this line that the result of a dictionary access is always an optional, so the compiler will need you to handle that possibility.

Reply:
rewrote it with a map, but i think because this is a three coloring program that enum is unlikely to change again and it would be ok to depend on